STEP/TypeVisitor.cs

```diff
@@ -179,17 +190,18 @@ public void Visit(ArrDclNode n)
     {
         DclVisitor dclVisitor = new DclVisitor(_symbolTable);
         n.Left.Accept(dclVisitor);
-        if (n.Right is null)
-        {
-            n.Type = n.Left.Type;
-            return;
-        }
-
+        n.Left.Type.ArrSize = n.Size;
+        n.Type = n.Left.Type;
+        if (n.Right is null) return;
         n.Right.Type = n.Left.Type;
         n.Right.Accept(this);
-        n.Type = (n.Left.Type != n.Right.Type)
-            ? throw new TypeException(n, $"Type mismatch, type of left: {n.Left.Type}, type of right: {n.Right.Type}")
-            : n.Left.Type;
+        if (n.Right is IdNode id) {
+            if (id.Type.IsArray) {
+                if (n.Left.Type.ArrSize < n.Right.Type.ArrSize) throw new TypeException(n, $"Array size mismatch, {n.Left.Id} can only fit {n.Left.Type.ArrSize} elements. {id.Id} has {id.Type.ArrSize} elements");
```

Comment:
hvorfor ikke bruge id.Type.ArrSize istedet for n.Right.Type.ArrSize?

Reply:
Du skal i hvert fald være konsistent, da du accesser den samme data på to forskellige måder her